### PR TITLE
Trev8939/find place

### PR DESCRIPTION
**Pull request by TADraeseke**
Sample which demonstrates how to find a place.

Uses Geocode, Suggestions, GeocodeParameters, etc.

### CODE REVIEW

**Comment by doneill** [comment]
@TADraeseke I can get the app to crash with the following exception: 

```
java.lang.IllegalArgumentException: Parameter searchText must not be null
    at com.esri.arcgisruntime.sample.findplace.MainActivity$8.run(MainActivity.java:460)
```

To reproduce: 
- Enter `Starbucks` in POI Text
- Enter/Select 'Portland, OR` in Proximity Text
- Tap `Redo Search in this Area` when returned to the Map.  
- Crash

File: find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,535 @@
...
+import com.esri.arcgisruntime.symbology.PictureMarkerSymbol;
+import com.esri.arcgisruntime.tasks.geocode.GeocodeParameters;
+import com.esri.arcgisruntime.tasks.geocode.GeocodeResult;
+import com.esri.arcgisruntime.tasks.geocode.LocatorTask;
+import com.esri.arcgisruntime.tasks.geocode.SuggestParameters;
+import com.esri.arcgisruntime.tasks.geocode.SuggestResult;
+
+public class MainActivity extends AppCompatActivity {
+
+  String[] reqPermissions =
```
Comment (TADraeseke) [inline find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java]: done

File: find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+      new String[] { Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION };
+  private int requestCode = 2;
```
Comment (TADraeseke) [inline find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java]: done

File: find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,535 @@
...
+  private String TAG;
```
Comment (TADraeseke) [inline find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java]: done

File: find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,535 @@
...
+  private SearchView mPoiSearchView;
+  private SearchView mLocationSearchView;
+  private String COLUMN_NAME_ADDRESS = "address";
```
Comment (TADraeseke) [inline find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java]: done

File: find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,535 @@
...
+  private String[] mColumnNames = { BaseColumns._ID, COLUMN_NAME_ADDRESS };
```
Comment (TADraeseke) [inline find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java]: done

File: find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,535 @@
...
+  private PictureMarkerSymbol mPinSourceSymbol;
+  private Geometry mCurrentExtentGeometry;
+  private Callout mCallout;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    TAG = "Find a place";
```
Comment (TADraeseke) [inline find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java]: done

File: find-place/src/main/res/layout/suggestion.xml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+              android:layout_width="match_parent"
+              android:layout_height="wrap_content"
+              android:orientation="horizontal">
+
+    <TextView
+            android:id="@+id/suggestion_address"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="5dp"
+            android:textSize="16dip"/>
```
Comment (TADraeseke) [inline find-place/src/main/res/layout/suggestion.xml]: done

File: find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java [hunk 8]
```diff
@@ -0,0 +1,535 @@
...
+    mPoiGeocodeParameters.setPreferredSearchLocation(mPreferredSearchLocation);
+    mPoiGeocodeParameters.setSearchArea(mPreferredSearchLocation);
+    // Execute async task to find the address
+    mLocatorTask.addDoneLoadingListener(new Runnable() {
+      @Override
+      public void run() {
+        if (mLocatorTask.getLoadStatus() == LoadStatus.LOADED) {
+          // Call geocodeAsync passing in an address
+          final ListenableFuture<List<GeocodeResult>> geocodeResultListenableFuture = mLocatorTask
+              .geocodeAsync(address, mPoiGeocodeParameters);
```
Comment (TADraeseke) [inline find-place/src/main/java/com/esri/arcgisruntime/sample/findplace/MainActivity.java]: Done

**Comment by TADraeseke** [comment]
@doneill Thanks for review, and particularly for capturing that crash. Crash and all comments addressed. Ready for second review.

File: find-place/src/main/res/layout/activity_main.xml [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+            android:queryHint="@string/search_hint"
+            android:layout_alignParentTop="true"
+            android:layout_centerHorizontal="true"
+            android:layout_marginTop="5dp"
+            android:layout_marginLeft="5dp"
+            android:layout_marginRight="5dp">
+    </android.support.v7.widget.SearchView>
+
+    <Button
+            android:text="Redo search in this area"
```
Comment (doneill) [inline find-place/src/main/res/layout/activity_main.xml]: Use `@string` resource

File: find-place/src/main/res/layout/activity_main.xml [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+            android:layout_width="fill_parent"
+            android:layout_height="fill_parent">
+    </com.esri.arcgisruntime.mapping.view.MapView>
+
+    <android.support.v7.widget.SearchView
+            android:id="@+id/location_searchView"
+            android:background="@color/colorWhiteSemiTransparent"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_weight="7"
```
Comment (doneill) [inline find-place/src/main/res/layout/activity_main.xml]: This layout param is obsolete and can be removed

File: find-place/src/main/res/layout/activity_main.xml [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+            android:layout_below="@+id/poi_searchView"
+            android:layout_alignParentStart="true">
+    </android.support.v7.widget.SearchView>
+
+    <android.support.v7.widget.SearchView
+            android:id="@+id/poi_searchView"
+            android:background="@color/colorWhiteSemiTransparent"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_weight="7"
```
Comment (doneill) [inline find-place/src/main/res/layout/activity_main.xml]: This layout param is obsolete and can be removed

**Comment by doneill** [comment]
@TADraeseke Looks good, just some additional minor changes to the layout, once updated you can merge.

**Pull request merged**